_posts/2017-07-03-Devoxx-pl.md

```diff
@@ -165,4 +165,203 @@ With tools like [Split](https://www.split.io) and [LaunchDarkly](https://launchd
+
+> Java 9 Release date: September 21st 2017
+
+### JShell
+
+JShell is the new REPL (Read-Eval-Print Loop) for Java.
+It can be used to run commands and get results immediately.
+For user-friendliness the semicolons can be omitted after the instructions in JShell.
+
+Example command:
+{% highlight Java %}
+jshell> List.of(1).getClass()
+$1 ==> class java.util.ImmutableCollections$List1
+{% endhighlight %}
+
+### Optionals 
+
+Several improvements will be added to the `Optional` class.
+Optionals can be turned into streams and have `filter`, `flatMap` and `map` methods.
+
+For eager evaluation these functional methods can be applied directly to the `Optional`.
+{% highlight Java %}
+jshell> Optional.of(1).map(x->x*3)
+$2 ==> Optional[3]
+{% endhighlight %}
+When using `stream()` in front of the functional methods a `ReferencePipeline` is returned, this can be used for lazy evaluation.
+{% highlight Java %}
+jshell> Optional.of(1).stream().map(x->x*3)
+$3 ==> java.util.stream.ReferencePipeline
+{% endhighlight %}
+
+An `or()` method will be added to chain a supplier to empty Optionals.
+{% highlight Java %}
+jshell> Optional.empty().or(()->Optional.of("Devoxx rocks!"))
+$4 ==> Optional[Devoxx rocks!]
+{% endhighlight %}
+
+### Streams
+
+Two new methods will be added to the `Stream` interface, `dropWhile` and `takeWhile`.
+For ordered streams these methods drop or take elements while the predicate is true.
+In unordered streams `dropWhile` returns a subset of elements starting from the first predicate match, `takeWhile` returns a subset of elements matching the predicate.
+
+{% highlight Java %}
+Stream<T> dropWhile​(Predicate<? super T> predicate)
+Stream<T> takeWhile​(Predicate<? super T> predicate)
+
+jshell> IntStream.range(1,10).takeWhile(x-> x<5).boxed().collect(Collectors.toList())
+$5 ==> [1, 2, 3, 4]
+
+jshell> IntStream.range(1,10).dropWhile(x-> x<5).boxed().collect(Collectors.toList())
+$6 ==> [5, 6, 7, 8, 9]
+{% endhighlight %}
+
+### Concurrency
+
+`CompletableFuture` will be extended with a `copy`.
+The copied CompletableFuture is a defensive copy and is prevented from completing the original.
+
+{% highlight Java %}
+jshell> CompletableFuture.runAsync(()->{while(true){}})
+$7 ==> java.util.concurrent.CompletableFuture@26aa12dd[Not completed]
+
+jshell> $7.copy()
+$8 ==> java.util.concurrent.CompletableFuture@711f39f9[Not completed]
+
+jshell> $8.cancel(true)
+$9 ==> true
+
+jshell> $8
+$8 ==> java.util.concurrent.CompletableFuture@711f39f9[Completed exceptionally]
+
+jshell> $7
+$7 ==> java.util.concurrent.CompletableFuture@26aa12dd[Not completed, 1 dependents]
+{% endhighlight %}
+
+A new `ProcessHandle` will be able to get information and control processes.
+
+### Bits and pieces
+
+The underscore will become a keyword, so assigning a value to _ does not work.
+This is probably a feature for the future where _ will be used for matching arguments of any type.
+Assigning a value to __ will keep on working.
```

Comment:
Mmm, I don't get the point of mentioning this then. :-)
If __ works, then _a, _someVariable, ... will also. Was there something special with double underscore?